### PR TITLE
Chainpack rpc block

### PR DESCRIPTION
**Pull request by fvacek**

### CODE REVIEW

File: src/bin/cp2cp.rs [hunk 1]
```diff
@@ -43,27 +47,46 @@ fn main() {
...
     };
 
+    let mut chainpack_rpc_block_header = None;
     let res = if opts.cpon_input {
         let mut rd = CponReader::new(&mut reader);
         rd.read()
     } else {
         let mut rd = ChainPackReader::new(&mut reader);
+        if opts.chainpack_rpc_block {
+            let length = rd.read_uint_data().unwrap();
```
Comment (j4r0u53k) [inline src/bin/cp2cp.rs:73]: `.expect("length should be uint");`

File: src/bin/cp2cp.rs [hunk 2]
```diff
@@ -43,27 +47,46 @@ fn main() {
...
+            let proto = rd.read().unwrap().as_int();
```
Comment (j4r0u53k) [inline src/bin/cp2cp.rs:74]: .expect("...");

File: src/bin/cp2cp.rs [hunk 3]
```diff
@@ -43,27 +47,46 @@ fn main() {
...
+        }
         rd.read()
     };
     let rv = match res {
         Err(e) => {
-            eprintln!("Parse input error: {:?}", e);
-            process::exit(1);
+            match e.reason {
+                ReadErrorReason::UnexpectedEndOfStream => {
+                    process::exit(2);
```
Comment (j4r0u53k) [inline src/bin/cp2cp.rs:83]: What is the meaning of exit code `2` (and `1` in the next match arm) here? Constants would be better.

Comment (syyyr) [reply src/bin/cp2cp.rs]: agree

**Pull request merged**